js/src/panelStack.js

```diff
@@ -0,0 +1,68 @@
+import * as React from "react";
+import PropTypes from "prop-types";
+import { PanelStack2 } from "@blueprintjs/core";
+import styles from "./PanelStack.module.css";
+import handlers from "./handlers";
+
+const PanelsType = PropTypes.arrayOf(
+  PropTypes.shape({
+    id: PropTypes.string,
+    title: PropTypes.string,
+    content: PropTypes.node,
+  })
+);
+
+const Panel = ({ content }) => {
+  return React.createElement("div", { style: { padding: 10 } }, content);
+};
+Panel.propTypes = {
+  content: PropTypes.node,
+};
+
+const PanelStack = ({ panels, ns, size, ...propsRest }) => {
+  const [stack, setStack] = React.useState([
+    {
+      renderPanel: Panel,
+      title: panels[0].title,
+      props: { content: panels[0].content },
+    },
+  ]);
+  const addToPanelStack = React.useCallback(
+    (newPanel) => setStack((stack) => [...stack, newPanel]),
+    []
+  );
+  const removeFromPanelStack = React.useCallback(
+    () => setStack((stack) => stack.slice(0, -1)),
+    []
+  );
+  handlers[ns] = {
+    openPanel: (panelId) => {
+      const { title, content } = panels.find(({ id }) => id == panelId);
+      addToPanelStack({
+        renderPanel: Panel,
+        title,
+        props: { content },
+      });
+    },
+    removePanel: removeFromPanelStack,
+  };
```

Comment:
This will modify a global variable each time the component is rerendered; also we are never cleaning it up.

If you have time left: Put this inside `useEffect()` and add proper cleanup.

Reply:
Done